packages/smooth_app/lib/database/barcode_product_query.dart

```diff
@@ -1,30 +1,25 @@
 import 'dart:async';
 
 import 'package:openfoodfacts/openfoodfacts.dart';
-import 'package:openfoodfacts/utils/CountryHelper.dart';
 import 'package:smooth_app/data_models/fetched_product.dart';
 import 'package:smooth_app/database/dao_product.dart';
 import 'package:smooth_app/database/product_query.dart';
 
 class BarcodeProductQuery {
   BarcodeProductQuery({
     required this.barcode,
-    required this.language,
-    required this.country,
     required this.daoProduct,
   });
 
   final String barcode;
-  final OpenFoodFactsLanguage? language;
-  final OpenFoodFactsCountry? country;
   final DaoProduct daoProduct;
 
   Future<FetchedProduct> getFetchedProduct() async {
     final ProductQueryConfiguration configuration = ProductQueryConfiguration(
       barcode,
       fields: ProductQuery.fields,
-      language: language,
-      country: country,
+      language: ProductQuery.getLanguage(),
+      country: ProductQuery.getCountry(),
```

Comment:
In `getProduct()`  we set `queryParameters: configuration.getParametersMap(),` and in `getParametersMap` we access the global values

Reply:
Thank you @M123-dev for your code review!
Your comment may be relevant, but I'm not confident enough in off-dart's evolutions to rely 100% on them.
It makes more sense in smoothie to keep all the API calls with the same parameters, specifically when not all API calls have the same parameters on the off-dart side.